lib/puppet/file_serving/metadata.rb

```diff
@@ -22,10 +22,16 @@ def checksum_type=(type)
     @checksum_type = type
   end
 
+  def source_permissions=(source_permissions)
+    raise(ArgumentError, "Unsupported source_permission #{source_permissions}") unless [:use, :use_when_creating, :ignore].include?(source_permissions.intern)
```

Comment:
@joshcooper ran into some weirdness here. `source_permissions` was being compared to the symbol array as a string, and so Puppet was throwing exceptions.. `intern'd` it for now, but I'm not sure I understand why that's happening (I verified `source_permissions` is a symbol itself).